#! /bin/sh /usr/share/dpatch/dpatch-run
## 12-gcc-4.3.dpatch by Michael Meskes <meskes@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Some changes needed by gcc-4.3

@DPATCH@

diff -Naurp virtualbox-ose-1.6.0-dfsg.orig/src/VBox/Frontends/VirtualBox/src/VBoxGlobal.cpp virtualbox-ose-1.6.0-dfsg/src/VBox/Frontends/VirtualBox/src/VBoxGlobal.cpp
--- virtualbox-ose-1.6.0-dfsg.orig/src/VBox/Frontends/VirtualBox/src/VBoxGlobal.cpp	2008-04-30 14:05:33.000000000 +0000
+++ virtualbox-ose-1.6.0-dfsg/src/VBox/Frontends/VirtualBox/src/VBoxGlobal.cpp	2008-05-09 09:31:08.000000000 +0000
@@ -33,6 +33,8 @@
 #include "VBoxRegistrationDlg.h"
 #endif
 
+#include <stdlib.h>
+
 #include <qapplication.h>
 #include <qmessagebox.h>
 #include <qpixmap.h>
diff -Naurp virtualbox-ose-1.6.0-dfsg.orig/src/VBox/Main/VMMDevInterface.cpp virtualbox-ose-1.6.0-dfsg/src/VBox/Main/VMMDevInterface.cpp
--- virtualbox-ose-1.6.0-dfsg.orig/src/VBox/Main/VMMDevInterface.cpp	2008-04-30 14:05:49.000000000 +0000
+++ virtualbox-ose-1.6.0-dfsg/src/VBox/Main/VMMDevInterface.cpp	2008-05-09 09:31:08.000000000 +0000
@@ -31,6 +31,7 @@
 #include <VBox/VBoxGuest.h>
 #include <VBox/shflsvc.h>
 #include <iprt/asm.h>
+#include <stdio.h>
 
 #ifdef VBOX_HGCM
 #include "hgcm/HGCM.h"
diff -Naurp virtualbox-ose-1.6.0-dfsg.orig/src/recompiler/Sun/op-validate.sed virtualbox-ose-1.6.0-dfsg/src/recompiler/Sun/op-validate.sed
--- virtualbox-ose-1.6.0-dfsg.orig/src/recompiler/Sun/op-validate.sed	2008-04-30 14:05:14.000000000 +0000
+++ virtualbox-ose-1.6.0-dfsg/src/recompiler/Sun/op-validate.sed	2008-05-09 09:31:08.000000000 +0000
@@ -63,6 +63,8 @@ s/^[[:blank:]]*ret[[:blank:]]*\n*[[:blan
 /\.Lfe[0-9][0-9]*:/d
 /\.LFE[0-9][0-9]*:/d
 /size[[:space:]]/d
+/p2align[[:space:]]/d
+/^[/#][[:space:]]0[[:space:]]\"\"[[:space:]]2[[:space:]]*$/d
 /^[/#]NO_APP[[:space:]]*$/d
 /^$/!b bad
 b end
